homeassistant/components/homekit/type_media_players.py

```diff
@@ -142,3 +170,182 @@ def update_state(self, new_state):
+                CHAR_VOLUME_SELECTOR, setter_callback=self.set_volume_step)
+
+        if CHAR_ACTIVE_IDENTIFIER in self.chars:
+            self.sources = self.hass.states.get(self.entity_id).attributes.get(
+                ATTR_INPUT_SOURCE_LIST)
+            if self.sources:
+                self.char_active_identifier = serv_tv.configure_char(
+                    CHAR_ACTIVE_IDENTIFIER,
+                    setter_callback=self.set_input_source)
+                for index, source in enumerate(self.sources):
+                    serv_input = self.add_preload_service(
+                        SERV_INPUT_SOURCE, [CHAR_IDENTIFIER, CHAR_NAME])
+
+                    serv_input.configure_char(
+                        CHAR_CONFIGURED_NAME, value=source)
+                    serv_input.configure_char(CHAR_NAME, value=source)
+                    serv_input.configure_char(
+                        CHAR_IDENTIFIER, value=index)
+                    serv_input.configure_char(
+                        CHAR_IS_CONFIGURED, value=True)
+                    input_type = 3 if "hdmi" in source.lower() else 0
+                    serv_input.configure_char(CHAR_INPUT_SOURCE_TYPE,
+                                              value=input_type)
+                    serv_input.configure_char(
+                        CHAR_CURRENT_VISIBILITY_STATE, value=False)
+                    serv_tv.add_linked_service(serv_input)
+                    _LOGGER.debug('%s: Added source %s.', self.entity_id,
+                                  source)
+
+    def set_on_off(self, value):
+        """Move switch state to value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set switch state for "on_off" to %s',
+                      self.entity_id, value)
+        self._flag_active = True
+        service = SERVICE_TURN_ON if value else SERVICE_TURN_OFF
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        self.call_service(DOMAIN, service, params)
+
+    def set_play_pause(self, value):
+        """Move switch state to value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set switch state for "play_pause" to %s',
+                      self.entity_id, value)
+        service = SERVICE_MEDIA_PLAY if value else SERVICE_MEDIA_PAUSE
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        self.call_service(DOMAIN, service, params)
+
+    def set_toggle_mute(self, value):
+        """Move switch state to value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set switch state for "toggle_mute" to %s',
+                      self.entity_id, value)
+        self._flag_toggle_mute = True
+        params = {ATTR_ENTITY_ID: self.entity_id,
+                  ATTR_MEDIA_VOLUME_MUTED: value}
+        self.call_service(DOMAIN, SERVICE_VOLUME_MUTE, params)
+
+    def set_volume_step(self, value):
+        """Send volume step value if call came from HomeKit."""
+        _LOGGER.debug('%s: Step volume by %s',
+                      self.entity_id, value)
+
```

Comment:
Do we need a flag for that?

Reply:
I believe this is the same as play/pause. I did add a flag for tracking volume level. I still need to be able to differentiate between media players with and without volume level.

Reply:
You were right here as well. I didn't know that there is not UI inside iOS to display the current volume. The flag isn't necessary then.